sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -925,7 +925,7 @@ class Analyzer(override val catalogManager: CatalogManager)
         if (metaCols.isEmpty) {
           node
         } else {
-          val newNode = addMetadataCol(node)
+          val newNode = node.mapChildren(addMetadataCol(_, metaCols.map(_.exprId).toSet))
```

Comment:
Why this only does `addMetadataCol` on children?

Reply:
The change is from this fix https://github.com/apache/spark/pull/39895

Reply:
This looks correct to me because we need to propagate metadata cols to this node and need to add metadata cols in its children. Does master branch have the same code here?

Reply:
Yes, master branch has the same code.

Reply:
Oh, there was follow up.